frontend/src/components/grantaccess/form/UserDomainSelect.vue

```diff
@@ -0,0 +1,46 @@
+<script setup lang="ts">
+import { computed } from 'vue';
+import RadioButton from 'primevue/radiobutton';
+import { UserType } from 'fam-app-acsctl-api';
+
+const domainOptions = { IDIR: UserType.I, BCEID: UserType.B };
+
+const props = defineProps({
+    domain: { type: String, default: UserType.I },
+});
+
+const emit = defineEmits(['change']);
+const computedDomain = computed({
```

Comment:
we use a [writable computed variable](https://vuejs.org/guide/essentials/computed#writable-computed) here, so whenever we read its value, it's get from props.domain. And whenever we change it's value, it will call the set() method and call our emit method "change". 

Emit a method basically means calling a method from parent. So whenever emit has been called, it will call the method we got from the parent component GrantAccess.vue, which is "userDomainChange()", and it will update the domain in the formData. And once formData.value.domain got changed, it will pass the domain as props again to the child UserDomainSelect component and rerender it.  

```
const userDomainChange = (selectedDomain: string) => {
    formData.value.domain = selectedDomain;
    formData.value.userId = '';
};

<UserDomainSelect
  :domain="formData.value.domain"
  @change="userDomainChange"
/>
```